### PR TITLE
feat(quotas): Add ability to use namespace in non-global quotas.

### PR DESCRIPTION
**Pull request by TBS1996**
https://github.com/getsentry/relay/issues/2716

With this PR, we can also use namespace in non-global quotas. 

The reason for this PR was the requirement to be able to ratelimit based on namespace per org, but in reality it's pretty flexible so we can do namespace per project as well.

### CODE REVIEW

**Review by Dav1dde — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]

File: relay-quotas/src/redis.rs
Comment (iker-barriocanal) [inline relay-quotas/src/redis.rs:422]: You're effectively having two loops in one, and splitting it increases readability:

```rs
for i in 0..5 {
  assert_eq!(
    rate_limits[0].reason_code,
    Some(ReasonCode::new("ns: None"))
  );
}
for i in 0..5 {
  assert_eq!(rate_limits, vec![]);
}
```
(Same with the loop below)

File: relay-quotas/src/redis.rs
Comment (iker-barriocanal) [inline relay-quotas/src/redis.rs:440]: These two loops are asserting validating two different features. Could we split them into two separate tests, as per [Relay guidelines](https://www.notion.so/sentry/Relay-Code-and-Architecture-4fac481cf27741aba54d39594f1cae58?pvs=4#fd14645580f848d9969d124a59cc3f7d) indicate?

> Keep unit tests constrained to a single concern. Do not test multiple things in one test, do not test sequences of actions. Write multiple tests for each of the concerns.

This makes identifying the root cause easier on test failures.

**Review by iker-barriocanal — APPROVED** [review]
LGTM. Please address the suggestions below before merging.

File: relay-quotas/src/redis.rs
Comment (TBS1996) [reply relay-quotas/src/redis.rs:440]: so I don't agree here, the two loops are there two assert that with and without namespace is counted differently, this is only possible if we have them both in the same test to ensure theyre both connected to the same redis instance. if you look below this test at `fn test_simple_quota` then you see the original test that checked the behaviour without namespace

File: relay-quotas/src/redis.rs
Comment (TBS1996) [reply relay-quotas/src/redis.rs:422]: hmm i get your point but it also makes it a bit less clear the separation of the loop with and without the namespace, and we'd also have to duplicate the let ratelimits = ... block more.

I agree it's not as clear as can be though, I uploaded some changes. 

1. added comments
2. the 'quota_limit' variable should be more clear
3. flipped the if/else statements

Comment (iker-barriocanal) [reply relay-quotas/src/redis.rs:422]: This is probably my personal opinion, and not going to block the PR for that, but this is one case in which I'm ok with some code duplication given I find less cognitive load and thus more readable.

**Review by iker-barriocanal — APPROVED** [review]

**Pull request merged**
